pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/service/api/Functions.java

```diff
@@ -42,7 +42,29 @@ void registerFunction(final String tenant,
                           final String functionPkgUrl,
                           final FunctionConfig functionConfig,
                           final String clientRole,
-                          AuthenticationDataHttps clientAuthenticationDataHttps);
+                          AuthenticationDataSource clientAuthenticationDataHttps);
+
+    @Deprecated
```

Comment:
On these deprecated methods, can we add a comment indicating which method should be used instead?